modules/storages/app/common/storages/peripherals/storage_interaction/authentication_strategies/strategy.rb

```diff
@@ -56,6 +56,10 @@ def with_token(token)
             @token = token
             self
           end
+
+          def ==(other)
```

Comment:
:yellow_circle: Aren't there "rules"/best practices that one should also overwrite the `hash` method when overwriting the `==` method? Right now what could happen (as far as I can tell) is this:

```ruby
a = AuthenticationStrategy.new
b = a.dup
puts a == b # Output: true
h = {}
h[a] = 1
h[b] = 2
puts h.size # Output: 2
```

Of course it should be a surprise that the hash stores two keys, because `a` and `b` are the same, but because rubies maps are hash maps, they rely on the hash first and foremost, to check object identity (code above is untested, this is from memory of why not overwriting `hash` can be a problem).

Reply:
You are right, it is indeed good practice and I totally forgot. At the time I just wanted my test to pass. XD